src/Features/Core/Portable/PullMemberUp/QuickAction/InterfacePullerWithQuickAction.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.Editing;
+using static Microsoft.CodeAnalysis.CodeActions.CodeAction;
+
+namespace Microsoft.CodeAnalysis.PullMemberUp.QuickAction
+{
+    internal class InterfacePullerWithQuickAction : AbstractMemberPullerWithQuickAction
+    {
+        private IMethodSymbol FilterGetterOrSetter(IMethodSymbol getterOrSetter)
+        {
+            if (getterOrSetter == null)
+            {
+                return getterOrSetter;
+            }
+            else
+            {
+                return getterOrSetter.DeclaredAccessibility == Accessibility.Public ? getterOrSetter : null;
+            }
+        }
+
+        internal override async Task<CodeAction> CreateAction(ISymbol memberSymbol)
+        {
+            var memberToPullUp = memberSymbol;
+            if (memberSymbol is IPropertySymbol propertyOrIndexerSymbol)
+            {
+                memberToPullUp = CodeGenerationSymbolFactory.CreatePropertySymbol(
+                    propertyOrIndexerSymbol,
+                    propertyOrIndexerSymbol.GetAttributes(),
+                    propertyOrIndexerSymbol.DeclaredAccessibility,
+                    DeclarationModifiers.From(propertyOrIndexerSymbol),
+                    propertyOrIndexerSymbol.ExplicitInterfaceImplementations,
+                    propertyOrIndexerSymbol.Name,
+                    propertyOrIndexerSymbol.IsIndexer,
+                    FilterGetterOrSetter(propertyOrIndexerSymbol.GetMethod), 
+                    FilterGetterOrSetter(propertyOrIndexerSymbol.SetMethod));
+
+            }
+            var options = new CodeGenerationOptions(generateMembers: false, generateMethodBodies: false);
+            var changedDocument = await CodeGenerationService.AddMembersAsync(
+                ContextDocument.Project.Solution,
+                TargetTypeSymbol,
+                new ISymbol[] { memberToPullUp },
+                options: options,
+                cancellationToken: _cancellationToken);
```

Comment:
this work may be expensive.  why do it up front?  The purpose of hte DocumentChangeAction is to defer the exensive work until necessary (either for preview, or for invoking).

Reply:
I think I finally get the idea of the task we want here. So it should be an callback operation which use small amount of memory (As you previous comments, the closure here shouldn't capture too many things), but could potentially slow since the user don't need it until they invoke it. Am I right?

Reply:
GOt it, Thank you